titus-common/src/main/java/com/netflix/titus/common/util/loadshedding/tokenbucket/ArchaiusTokenBucketAdmissionController.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.common.util.loadshedding.tokenbucket;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.titus.common.framework.scheduler.ExecutionContext;
+import com.netflix.titus.common.framework.scheduler.ScheduleReference;
+import com.netflix.titus.common.framework.scheduler.model.ScheduleDescriptor;
+import com.netflix.titus.common.runtime.TitusRuntime;
+import com.netflix.titus.common.util.loadshedding.AdmissionController;
+import com.netflix.titus.common.util.loadshedding.AdmissionControllerRequest;
+import com.netflix.titus.common.util.loadshedding.AdmissionControllerResponse;
+import com.netflix.titus.common.util.retry.Retryers;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A wrapper around {@link TokenBucketAdmissionController} which dynamically loads configuration from Archaius.
+ * When configuration changes, the current {@link TokenBucketAdmissionController} instance is discarded, a new one
+ * is created. As a result token buckets state is reset, which is fine as long as changes do not happen too
+ * frequently.
+ */
+public class ArchaiusTokenBucketAdmissionController implements AdmissionController, Closeable {
```

Comment:
I think it would be better to make this a generic configuration loader implementation and put the archaius part  behind a Configuration abstraction as there isn't really anything Archaius specific in this class.